### PR TITLE
Support to update disk/network offering tags from UI

### PR DESCRIPTION
**Pull request by slavkap**
### Description

added support to update tags of disk and network offerings from UI
fix for delete the last tag

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### How Has This Been Tested?
manually tested

### CODE REVIEW

**Review by rhtyd — APPROVED** [review]
LGTM - did n't test it

**Comment by rhtyd** [comment]
Please review/test @Pearl1594 @shwstppr @davidjumani @Spaceman1984

File: ui/src/views/AutogenView.vue [hunk 1]
```diff
@@ -983,13 +983,13 @@ export default {
...
             if (!input === undefined || input === null ||
-              (input === '' && !['updateStoragePool', 'updateHost', 'updatePhysicalNetwork'].includes(action.api))) {
+              (input === '' && !['updateStoragePool', 'updateHost', 'updatePhysicalNetwork', 'updateDiskOffering', 'updateNetworkOffering'].includes(action.api))) {
               if (param.type === 'boolean') {
                 params[key] = false
               }
               break
             }
-            if (!input) {
+            if (!input && !['tags'].includes(key)) {
```
Comment (utchoang) [inline ui/src/views/AutogenView.vue:992]: I think it should be `key !== 'tags'`

**Review by shwstppr — APPROVED** [review]
Tested.  LGTM

File: ui/src/views/AutogenView.vue (hunk 1 above)
Comment (shwstppr) [reply ui/src/views/AutogenView.vue:992]: @utchoang should we re-look this behaviour of skipping params that do not have a value in API request.
It should be already handled by the `required` property of the parameter and there can be some parameters that may allow passing empty text.
`!['tags'].includes(key)` should work same as `key !== 'tags`?

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by Pearl1594 — APPROVED** [review]
Verified. LGTM

File: ui/src/views/AutogenView.vue (hunk 1 above)
Comment (slavkap) [reply ui/src/views/AutogenView.vue:992]: @utchoang I left it like this `!['tags'].includes(key)` for future fixes/enhancements if somebody wants to add parameters that could be an empty text. If this is not acceptable, I could change it with your suggestion

**Review by utchoang — APPROVED** [review]

**Pull request merged**